metron-deployment/roles/metron_streaming/tasks/main.yml

```diff
@@ -51,4 +51,8 @@
 
 - include: hdfs_purge.yml
 
+- include: create_index_templates.yml
+  when: install_elasticsearch == True
```

Comment:
One way to do it that might result in some cleaner code in multiple places, is to get rid of the generic `search` host group.  In its place, we define a `solr` host group when we want Solr installed and an `elasticsearch` host group when we want Elasticsearch installed.  

You then extract this index creation functionality into a separate role `metron-elasticsearch` (or something like that).  You can then target this functionality at hosts using the `elasticsearch` host group.  This ensures it will not run when a cluster is using Solr.

This follows the pattern being used in METRON-211 where we have `metron-kafka` and `metron-hbase`.  It also gives us greater granularity in targeting this functionality to hosts versus an uber-role like `metron-streaming` that does a ton of different things.  Even though those different things are all related to the same end goal; deploying Metron streaming functionality.